clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -327,6 +327,41 @@ void Sema::CheckDeprecatedSYCLAttributeSpelling(const ParsedAttr &A,
     return;
   }
 
+  // Diagnose SYCL 2017 spellings in later SYCL modes.
+  if (LangOpts.getSYCLVersion() > LangOptions::SYCL_2017) {
+    // All attributes in the cl vendor namespace are deprecated in favor of a
+    // name in the sycl namespace as of SYCL 2020.
+    if (A.hasScope() && A.getScopeName()->isStr("cl")) {
+      DiagnoseDeprecatedAttribute(A, "sycl", NewName);
+      return;
+    }
+
+    // All GNU-style spellings are deprecated in favor of a C++-style spelling.
```

Comment:
Mention this as well please. I assume this and the cl namespace deprecation is part of SYCL2020 spec as mentioned in commit title but I think its nice having the details in commit message to make it easier for people not familiar with spec

Reply:
That's also already mentioned (I listed the spellings we deprecated explicitly). LMK if you think there's more work for me to do in the commit message though!

Reply:
Oh I see. I didn't realize we just had the explicitly mentioned spellings in those namespaces. My bad!